improve: readme infos

kokekanon opened the pull request:
note: when modifying the readme I realized that some comments were missing and added the shop server part

My part is ready, someone needs to modify this information that I'm unaware of.

| missing element                            	|
|--------------------------------------------	|
| remove or integrate new gifs               	|
| add Roadmap information                    	|
| confirm information of "Support Protocol". 	|
| integrate this [wiki Attached Effects](https://github.com/Nottinghster/otclient/wiki/Wiki-Attached-Effects) <br>~~I did it about 3 months ago, I no longer have access to it to improve it.~~<br> to the main repo  wiki     	|



# 1) improve: Readme

# demo :

https://github.com/kokekanon/otclient.readme/tree/readme3


requested by mehah about 3 months ago haha


### Screenshot:
....
-----------
....

![image](https://github.com/mehah/otclient/assets/114332266/b96af9be-594e-4c9a-ad74-de66433e49b6)

....
-----------
....

![test](https://github.com/mehah/otclient/assets/114332266/7ba44ae6-7e7b-40ba-8646-4b3f854e6299)

....

-----------

![image](https://github.com/mehah/otclient/assets/114332266/d508c19d-bec2-46c7-ad91-e85f6c8a3483)

....
-----------
....

![image](https://github.com/mehah/otclient/assets/114332266/2299caed-01b2-4167-8274-9b994964daf1)

....
-----------


# 2) Add comments 

| add comments                                                                                                                                                                                      	| file                                             	|
|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------	|--------------------------------------------------	|
| example attachParticleEffect: ```g_game.getLocalPlayer():attachParticleEffect("creature-effect") ```                                                                                                                    	| modules\game_attachedeffects\attachedeffects.lua 	|
| Add comment to minimap  tfs c++ (old) : <br>```void ProtocolGame::sendWorldTime()```<br>tfs lua (new) :<br> ```function Player.sendWorldTime(self, time)```<br>Canary: <br>```void ProtocolGame::sendTibiaTime(int32_t time`) ```	| modules\game_minimap\minimap.lua ![bbb](https://github.com/mehah/otclient/assets/114332266/3717e350-adf0-4cec-b62c-4a87b2df1c46)  |




# 3) Add shop v8 server-side
 
![image](https://github.com/mehah/otclient/assets/114332266/1388af50-6955-48df-9cef-4d208aef4770)
test in clean version otc and tfs 1.5 8.6

kokekanon committed:
README.md Parte1

kokekanon committed:
parte2

kokekanon committed:
parte3

kokekanon committed:
video

kokekanon committed:
fix link broken config.h

kokekanon committed:
Update README.md

kokekanon committed:
Update README.md

kokekanon committed:
Update README.md

kokekanon committed:
Update README.md

kokekanon committed:
Update README.md

kokekanon committed:
Update README.md

kokekanon committed:
Add comments

example attachParticleEffect: g_game.getLocalPlayer():attachParticleEffect("creature-effect") 	modules\game_attachedeffects\attachedeffects.lua
delete unnecessary print
print("No stats bars dimensions found: ", dimension, " on reloadCurrentTab()")	modules\game_interface\widgets\statsbar.lua
Add comment to minimap tfs c++ (old) :
void ProtocolGame::sendWorldTime()
tfs lua (new) :
function Player.sendWorldTime(self, time)
Canary:
void ProtocolGame::sendTibiaTime(int32_t time`) 	modules\game_minimap\

kokekanon committed:
3) Add shop v8 server-side

test in clean version otc and tfs 1.5 8.6

kokekanon committed:
Merge branch 'readme3' of https://github.com/kokekanon/otclient.readme into readme3

kokekanon committed:
Update README.md

matheussf commented on modules/game_interface/widgets/statsbar.lua:
```diff
@@ -479,7 +479,7 @@ function StatsBar.reloadCurrentTab()
         return constructStatsBar(dimension, currentStats.placement)
     else 
-        print("No stats bars dimensions found: ", dimension, " on reloadCurrentTab()")
+        -- print("No stats bars dimensions found: ", dimension, " on reloadCurrentTab()")
```
I wonder why is this unnecessary?
I've added this because if someone is adding new stats bars it might fall into this line and help development debug.
Thinking of that, I wonder if there's a separate way to print only for debug

kokekanon replied:
![image](https://github.com/mehah/otclient/assets/114332266/4658b4f4-8e9e-4106-843b-31a319406c80)
I don't want to see the bar,I voluntarily hide it
logout/ login
and all those prints came up.

matheussf replied:
Hmm it seems to be a bug, can you create an issue about this? I'll fix it.

kokekanon replied:
undo

matheussf reviewed (commented):
This new readme with gifs/videos for context is insane, thanks!
Awesome work!

kokekanon committed:
undo "No stats bars dimensions found: ", dimens"

kokekanon committed:
Merge branch 'readme3' of https://github.com/kokekanon/otclient.readme into readme3

mehah commented:
@kokekanon 

**current:**
![image](https://github.com/mehah/otclient/assets/2267386/7f1a5f2f-f064-4ff7-8322-0b0527d22a4f)

**new readme (fix)**
![image](https://github.com/mehah/otclient/assets/2267386/b99b9ec8-7c7c-4de8-9c37-9f247e88d493)

NICE WORK!!! :heartpulse:

kokekanon committed:
fix

kokekanon committed:
video Creature Information By UIWidget

kokekanon committed:
Update README.md

kokekanon committed:
Update README.md

kokekanon committed:
Update README.md

mehah commented:
@kokekanon 

If you find it necessary, you can clean up the features, there are many things that are unnecessary to inform and only make it more polluted.

type: c++20, Optimized Terminal, Idle Animation Support, Some bugs fixed contained in edubart/otclient etc...

you can replace it with text saying this client contains XYZ optimizations compared to otc edubart,

Unfortunately, I'm terrible at this, if not, it would help you to describe it.

kokekanon committed:
Merge remote-tracking branch 'upstream/main' into readme3

kokekanon committed:
test

kokekanon committed:
Update README.md

kokekanon commented:
Like this?
there are several of your videos on youtube in the otland post, which I can put there, along with a brief description.

![image](https://github.com/mehah/otclient/assets/114332266/c6b92b0d-0cbc-42e0-80b1-63a3e405c9d7)

-----------------

drop-down menu

![asi](https://github.com/mehah/otclient/assets/114332266/d134b5c8-bc6d-4f33-8850-1ce84b8145db)


moved to drop-down menu :
- C++20
- Floor Shadowing



delete :
- Crosshair
- Optimized Terminal
-  Some bugs fixed contained in [edubart/otclient](https://github.com/edubart/otclient)
- Idle Animation Support

Posible delete? or moved to drop-down menu ?:
- Floating Effects Option , 
- Refactored Walk System

kokekanon committed:
Update README.md

mehah commented:
@kokekanon 
yeap :)

kokekanon committed:
fix

kokekanon committed:
Update README.md

kokekanon committed:
Update README.md

rodrigopaixaorj committed:
Merge branch 'main' into readme3

kokekanon committed:
Update README.md

mehah commented:
@kokekanon 

I think it's better to put Support HTML/CSS syntax, because it doesn't support all the features of html/css.

https://github.com/user-attachments/assets/3fa1803a-2759-4b2f-890e-05f987f43260

There's this video too if you want to post it.

kokekanon committed:
Update README.md

leoddias commented:
@kokekanon Awesome work!

mehah committed:
Merge branch 'main' into readme3

kokekanon committed:
1/2

kokekanon committed:
Update README.md

kokekanon committed:
2/2

kokekanon commented:
wiki Attached Effects https://github.com/user-attachments/files/16553549/README.md

mehah committed:
Merge branch 'main' into readme3

matheussf reviewed (approved):
LGTM! Just a quick suggestion for better Git flow in future PRs: it might be more effective to keep PRs smaller and focused on specific contexts. For example, in this case, you could have one PR for the readme.txt and another for the shopSERVER. Working on each task separately and opening individual PRs for each would make the review process less tedious and time-consuming. Using Git Flow could be a good approach—create a branch for the feature and then merge the sub-tasks into this feature branch.
